app/src/main/java/xyz/jhughes/laundry/apiclient/MachineAPI.java

```diff
@@ -1,39 +1,45 @@
 package xyz.jhughes.laundry.apiclient;
 
-import java.util.ArrayList;
+
 import java.util.List;
 import java.util.Map;
 
 import retrofit2.Call;
-import retrofit2.http.GET;
-import retrofit2.http.Path;
-import xyz.jhughes.laundry.LaundryParser.Locations;
+import xyz.jhughes.laundry.BuildConfig;
+import xyz.jhughes.laundry.LaundryParser.LocationResponse;
 import xyz.jhughes.laundry.LaundryParser.Machine;
 import xyz.jhughes.laundry.LaundryParser.MachineList;
 
-/**
- * Retrofit interface for the Machine API.
- */
-public interface MachineAPI {
-    @GET("/v2/location/{location}")
-    Call<ArrayList<Machine>> getMachineStatus(
-            @Path("location") String location
-    );
-
-    @GET("/v2/location/all")
-    Call<Map<String,MachineList>> getAllMachines();
-
-    @GET("/v2/locations")
-    Call<List<Locations>> getLocations();
-
-    @GET("/v2-debug/location/{location}")
-    Call<ArrayList<Machine>> getMachineStatus_DEBUG(
-            @Path("location") String location
-    );
-
-    @GET("/v2-debug/location/all")
-    Call<Map<String,MachineList>> getAllMachines_DEBUG();
-
-    @GET("/v2-debug/locations")
-    Call<List<Locations>> getLocations_DEBUG();
+public class MachineAPI {
+    public static final String API_ROOT = "http://laundry-api.sigapp.club";
+    private final boolean DEBUG = BuildConfig.DEBUG;
+    private MachineService machineService;
+
+    public MachineAPI(MachineService machineService) {
+        this.machineService = machineService;
+    }
+
+    public Call<List<LocationResponse>> getLocations() {
+        if (DEBUG) {
```

Comment:
hmmmm I'm guessing there's a much cleaner way of doing this debug thing...i'll look into it in the next couple days

Reply:
More than likely. I just cleaned it up the best i could

Reply:
@piticent123 We might be able to inject a debugging variable using dagger